src/query/encoding.ts

```diff
@@ -325,5 +330,5 @@ export function scaleType(fieldQ: FieldQuery) {
     timeUnit: timeUnit as TimeUnit,
     bin: bin as BinParams
   };
-  return compileScaleType({type: scale.type}, channel, fieldDef, markType);
+  return compileScaleType({type: scale.type}, channel as Channel, fieldDef, markType);
```

Comment:
channel can't be row or column anymore. Why do we need to force the type now?

Reply:
`channel` is of type ExtendedChannel rather than of type `Channel`, which clashes with the expected type of Channel. Should I change the variable names to be extendedChannel instead of channel?

Reply:
Vega-Lite used to only have Channel but now we distinguish between Channel and ExtendedChannel (Channel + row, column, facet). I think CompassQL could support extended channels unless it doesn't make sense for the particular functionality.

What do you think?

Reply:
ExtendedChannels make the most sense to me for Compassql. The issue I'm having is that I am not sure how to adapt the code to work with the vl functions that expect Channels. It seems that if we've ruled out row, column, and facet as possibilities, an ExtendedChannel just becomes a Channel. Can TypeScript figure this out automatically? Am I just missing some checks in my code?